### ISSUE TITLE
False positive when multiplying against a negative duration

### ISSUE THREAD

**Issue by guidoiaquinti** [charithe/durationcheck#17]
**Sample code**
```
package dur

import "time"

const _ = -time.Hour * time.Duration(1)

const _ = time.Hour * -time.Duration(1)

const _ = -(time.Hour * time.Duration(1))
```

**Output**
```
Multiplication of durations: `time.Hour * -time.Duration(1)` (durationcheck)
const _ = time.Hour * -time.Duration(1)
          ^
```

**Comment by charithe** [comment]
Just to clarify, is this issue about the second case (`time.Hour * - time.Duration(1)`)? The linter does not flag the other two. 

The first and the second cases are very odd ways of writing this expression. I know they are technically valid, but, that's because of the weak type system in Go. The purpose of this linter is to catch syntactically correct but possibly erroneous ways in which durations can be used in a codebase. I am inclined to not try and fix the linter warning for the case you highlighted because there are better, much more readable ways of writing the same expression (`time.Hour * time.Duration(-1)` or ` -(time.Hour * time.Duration(1))`).
